src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -946,6 +946,178 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
+            string expectedErrorMessage)
+        {
+            const string CUSTOM_CONFIG = "custom-config.json";
+            TestHelper.ConstructNewConfigWithSpecifiedHostMode(CUSTOM_CONFIG, HostModeType.Production, TestCategory.POSTGRESQL);
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(requestType);
+                HttpRequestMessage request;
+                if (requestType is RestMethod.Get || requestType is RestMethod.Delete)
+                {
+                    request = new(httpMethod, requestPath);
+                }
+                else
+                {
+                    request = new(httpMethod, requestPath)
+                    {
+                        Content = JsonContent.Create(requestBody)
+                    };
+                }
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                string body = await response.Content.ReadAsStringAsync();
+                Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+                Assert.IsTrue(body.Contains(expectedErrorMessage));
+            }
+        }
+
+        /// <summary>
+        /// Validates the error message that is returned for REST requests with incorrect parameter type
+        /// when the engine is running in Production mode. The error messages in Production mode is
+        /// very generic to not reveal information about the underlying database objects backing the entity.
+        /// This test runs against a MsSql database.
+        /// </summary>
+        /// <param name="requestType">Type of REST request</param>
+        /// <param name="requestPath">Endpoint for the REST request</param>
+        /// <param name="expectedErrorMessage">Right error message that should be shown to the end user</param>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(RestMethod.Get, "/api/Book/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a table in production mode for MsSql database")]
+        [DataRow(RestMethod.Get, "/api/books_view_all/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a view in production mode for MsSql")]
+        [DataRow(RestMethod.Get, "/api/GetBook?id=one", TestHelper.REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request on a stored-procedure with incorrect parameter type in production mode for MsSql")]
```

Comment:
I think we should add a test for Graphql

Reply:
For GraphQL requests, when an incorrect parameter type is used, the execution does not occur till this point. Hotchocolate catches this and throws an error with the message `The specified argument value does not match the argument type`  when running in both development and production mode. We have a test `TestStoredProcedureQueryWithInvalidArgumentType` already that validates this error message